Codeinwp/otter-blocks
# Add Stripe Form Field & Temporary Submission Saving

Closes #.


### Summary


This combines the PRs https://github.com/Codeinwp/otter-blocks/pull/1778 and https://github.com/Codeinwp/otter-blocks/pull/1769

### Screenshots 

![image](https://github.com/Codeinwp/otter-blocks/assets/17597852/92461bc2-fc36-424f-a3ef-22c3f6660179)

![image](https://github.com/Codeinwp/otter-blocks/assets/17597852/dff79010-a5d0-4314-8052-1e68cbab19fc)


----

### Test instructions


⚠️ You will need an Otter Pro Key and a Stripe account (test mode) with some products.

The flow of the Stripe Field

1. You a Form with some Stripe fields (you can as many as you want)
2. When you press Submit, you will be redirected to the Stripe Checkout Session page (similar to Stripe Checkout Block)
3. Before completing the Card form, go to the Otter Submission page in another browser tab. You will see the post Marked as a `Draft` -- more about this behavior: https://github.com/Codeinwp/otter-blocks/pull/1769.
4. Go back to the Stripe session and complete the payment with some dummy data.
5. After completing the payment, you will be redirected to the same page, but the URL will have a special param called `stripe_checkout`. That is the ID of the session, and it will be used to determine if a user has paid or not.
6. If you see the Success message and go back to Otter Submission, you will see that the submission is no longer a `Draft`

ℹ️ If you enable webhook and autoresponder, while in the paying stage, those should not trigger. They must trigger *ONLY* if the payment is `paid` (you check this in Stripe Dashboard marked as Complete).

⚠️ To confirm the Submission, you must be on the redirect that Stripe will offer to you. If not, the submission will stay as a Draft.

ℹ️ If the payment is marked as done, but you did not visit the special URL. We have a corn job that runs every hour (this value is still in discussion) and checks if there are Submissions in `Draft` mode but with complete payment. If it detects such cases, it will trigger the confirmation process and the remaining functions (webhooks, autoresponder).

You must test if:

- The Stripe Checkout Page has the same products listed in the Form. If not, this is a sync issue, and you will need to provide the test instance.
- The Webhook, autoresponder, etc trigger before the payment is finished.

⚠️ Have in mind the other Form issues that have been reported: #1802 #1804, which will be solved in different PR



---- 

### Checklist before the final review
